mbedtls/src/pk/dsa/mod.rs

```diff
@@ -218,8 +218,12 @@ fn sample_secret_value<F: Random>(upper_bound: &Mpi, rng: &mut F) -> Result<Mpi>
 }
 
 fn encode_dsa_signature(r: &Mpi, s: &Mpi) -> Result<Vec<u8>> {
-    let r = BigUint::from_bytes_be(&r.to_binary()?);
-    let s = BigUint::from_bytes_be(&s.to_binary()?);
+    serialize_signature(&r.to_binary()?, &s.to_binary()?)
+}
+
+pub fn serialize_signature(r: &[u8], s: &[u8]) -> Result<Vec<u8>> {
```

Comment:
This function only outputs a correct DSA signature if `r` and `s` are the _r_ and _s_ components of a DSA signature (that part is obvious from the context), encoded as specified in [https://datatracker.ietf.org/doc/html/rfc8017#section-4.1](https://datatracker.ietf.org/doc/html/rfc8017#section-4.1). I think it would be good to add that as rust documentation on this function and perhaps a similar rustdoc on `deserialize_signature()`.